<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Genome Wide Associated Studies (GWAS)" | A Minimal Book Example</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Genome Wide Associated Studies (GWAS)" | A Minimal Book Example" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Genome Wide Associated Studies (GWAS)" | A Minimal Book Example" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="Yihui Xie" />


<meta name="date" content="2021-04-16" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="differential-expression-analysis-using-basic-r.html"/>
<link rel="next" href="hi-c-workflow.html"/>
<script src="libs/header-attrs-2.7/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a></li>
<li class="chapter" data-level="2" data-path="analyis-of-new-generation-sequencing-data-in-r.html"><a href="analyis-of-new-generation-sequencing-data-in-r.html"><i class="fa fa-check"></i><b>2</b> Analyis of new generation sequencing data in R</a>
<ul>
<li class="chapter" data-level="2.1" data-path="analyis-of-new-generation-sequencing-data-in-r.html"><a href="analyis-of-new-generation-sequencing-data-in-r.html#get-fastq-from-sra"><i class="fa fa-check"></i><b>2.1</b> Get fastq from SRA</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="get-fastq-from-sra-database.html"><a href="get-fastq-from-sra-database.html"><i class="fa fa-check"></i><b>3</b> Get fastq from SRA database</a>
<ul>
<li class="chapter" data-level="3.1" data-path="get-fastq-from-sra-database.html"><a href="get-fastq-from-sra-database.html#get-fastq-from-sra-database-1"><i class="fa fa-check"></i><b>3.1</b> Get fastq from SRA database</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="differential-expression-analysis-using-basic-r.html"><a href="differential-expression-analysis-using-basic-r.html"><i class="fa fa-check"></i><b>4</b> Differential expression analysis using basic R</a>
<ul>
<li class="chapter" data-level="4.1" data-path="differential-expression-analysis-using-basic-r.html"><a href="differential-expression-analysis-using-basic-r.html#gene-expression-analysis-of-histone-deacetylase-1-hdac1-knockout-mouse."><i class="fa fa-check"></i><b>4.1</b> Gene expression analysis of histone deacetylase 1 (HDAC1) knockout mouse.</a></li>
<li class="chapter" data-level="4.2" data-path="differential-expression-analysis-using-basic-r.html"><a href="differential-expression-analysis-using-basic-r.html#sources"><i class="fa fa-check"></i><b>4.2</b> Sources</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="genome-wide-associated-studies-gwas.html"><a href="genome-wide-associated-studies-gwas.html"><i class="fa fa-check"></i><b>5</b> Genome Wide Associated Studies (GWAS)"</a>
<ul>
<li class="chapter" data-level="5.1" data-path="genome-wide-associated-studies-gwas.html"><a href="genome-wide-associated-studies-gwas.html#snp-analysis-using-snpasoc-r-package"><i class="fa fa-check"></i><b>5.1</b> SNP analysis using SNPasoc R package</a></li>
<li class="chapter" data-level="5.2" data-path="genome-wide-associated-studies-gwas.html"><a href="genome-wide-associated-studies-gwas.html#gwas-using-plink"><i class="fa fa-check"></i><b>5.2</b> GWAS using PLINK</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="hi-c-workflow.html"><a href="hi-c-workflow.html"><i class="fa fa-check"></i><b>6</b> Hi-C workflow</a>
<ul>
<li class="chapter" data-level="6.1" data-path="hi-c-workflow.html"><a href="hi-c-workflow.html#using-straw-tool-to-extract-contact-matrix-from-.hic-files"><i class="fa fa-check"></i><b>6.1</b> Using straw tool to extract contact matrix from .hic files</a></li>
<li class="chapter" data-level="6.2" data-path="hi-c-workflow.html"><a href="hi-c-workflow.html#hic-pro-pipeline"><i class="fa fa-check"></i><b>6.2</b> HiC-Pro pipeline</a></li>
<li class="chapter" data-level="6.3" data-path="hi-c-workflow.html"><a href="hi-c-workflow.html#multihic-compare-for-compare-of-2-hi-c-datasets-rao-et-al.-2017"><i class="fa fa-check"></i><b>6.3</b> multiHiC compare for compare of 2 Hi-C datasets (Rao et al.Â 2017)</a></li>
<li class="chapter" data-level="6.4" data-path="hi-c-workflow.html"><a href="hi-c-workflow.html#joint-normalization-of-hi-c"><i class="fa fa-check"></i><b>6.4</b> Joint normalization of Hi-C</a></li>
<li class="chapter" data-level="6.5" data-path="hi-c-workflow.html"><a href="hi-c-workflow.html#difference-detection"><i class="fa fa-check"></i><b>6.5</b> Difference detection</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A Minimal Book Example</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="genome-wide-associated-studies-gwas" class="section level1" number="5">
<h1><span class="header-section-number">Chapter 5</span> Genome Wide Associated Studies (GWAS)"</h1>
<div id="snp-analysis-using-snpasoc-r-package" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> SNP analysis using SNPasoc R package</h2>
<p>Example demonstrate an association test for an illness for one single SNP.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="genome-wide-associated-studies-gwas.html#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;SNPassoc&quot;</span>)</span>
<span id="cb45-2"><a href="genome-wide-associated-studies-gwas.html#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;SNPassoc&quot;</span>)</span>
<span id="cb45-3"><a href="genome-wide-associated-studies-gwas.html#cb45-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(SNPs)</span>
<span id="cb45-4"><a href="genome-wide-associated-studies-gwas.html#cb45-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(SNPs)</span>
<span id="cb45-5"><a href="genome-wide-associated-studies-gwas.html#cb45-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(SNPs.info.pos)</span>
<span id="cb45-6"><a href="genome-wide-associated-studies-gwas.html#cb45-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-7"><a href="genome-wide-associated-studies-gwas.html#cb45-7" aria-hidden="true" tabindex="-1"></a><span class="co"># select 6-40 SNP and create SNP object</span></span>
<span id="cb45-8"><a href="genome-wide-associated-studies-gwas.html#cb45-8" aria-hidden="true" tabindex="-1"></a>mySNP <span class="ot">&lt;-</span> <span class="fu">setupSNP</span>(SNPs, <span class="dv">6</span><span class="sc">:</span><span class="dv">40</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb45-9"><a href="genome-wide-associated-studies-gwas.html#cb45-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-10"><a href="genome-wide-associated-studies-gwas.html#cb45-10" aria-hidden="true" tabindex="-1"></a><span class="co"># casco - 1 for case, 0 for control)</span></span>
<span id="cb45-11"><a href="genome-wide-associated-studies-gwas.html#cb45-11" aria-hidden="true" tabindex="-1"></a>mySNP</span>
<span id="cb45-12"><a href="genome-wide-associated-studies-gwas.html#cb45-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-13"><a href="genome-wide-associated-studies-gwas.html#cb45-13" aria-hidden="true" tabindex="-1"></a><span class="co"># association test</span></span>
<span id="cb45-14"><a href="genome-wide-associated-studies-gwas.html#cb45-14" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">association</span>(casco<span class="sc">~</span>sex<span class="sc">+</span>snp10001<span class="sc">+</span>blood.pre, <span class="at">data =</span> mySNP,</span>
<span id="cb45-15"><a href="genome-wide-associated-studies-gwas.html#cb45-15" aria-hidden="true" tabindex="-1"></a>                     <span class="at">model.interaction =</span> <span class="fu">c</span>(<span class="st">&quot;dominant&quot;</span>,<span class="st">&quot;codominant&quot;</span>))</span>
<span id="cb45-16"><a href="genome-wide-associated-studies-gwas.html#cb45-16" aria-hidden="true" tabindex="-1"></a>res</span></code></pre></div>
<p>0 - control sample size<br />
% - percent for each variant<br />
1 - case sample size<br />
% - percent for each varian<br />
OR - odd ratio<br />
lower/upper - 95% confidence interval for odd ratio<br />
p-value of likelihood ratio test<br />
AIC - Akaike Information Criterion</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="genome-wide-associated-studies-gwas.html#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># association scan for SNPs - separately for all models</span></span>
<span id="cb46-2"><a href="genome-wide-associated-studies-gwas.html#cb46-2" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">WGassociation</span>(protein, <span class="at">data =</span> mySNP, <span class="at">model =</span> <span class="st">&#39;all&#39;</span>) <span class="co"># same formula as protein~1,</span></span>
<span id="cb46-3"><a href="genome-wide-associated-studies-gwas.html#cb46-3" aria-hidden="true" tabindex="-1"></a><span class="co"># p-values for dominant model</span></span>
<span id="cb46-4"><a href="genome-wide-associated-studies-gwas.html#cb46-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dominant</span>(res)</span>
<span id="cb46-5"><a href="genome-wide-associated-studies-gwas.html#cb46-5" aria-hidden="true" tabindex="-1"></a><span class="co"># p-values for recessive model</span></span>
<span id="cb46-6"><a href="genome-wide-associated-studies-gwas.html#cb46-6" aria-hidden="true" tabindex="-1"></a><span class="fu">recessive</span>(res)</span>
<span id="cb46-7"><a href="genome-wide-associated-studies-gwas.html#cb46-7" aria-hidden="true" tabindex="-1"></a><span class="co"># complete statistics</span></span>
<span id="cb46-8"><a href="genome-wide-associated-studies-gwas.html#cb46-8" aria-hidden="true" tabindex="-1"></a><span class="fu">WGstats</span>(res)</span>
<span id="cb46-9"><a href="genome-wide-associated-studies-gwas.html#cb46-9" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res)</span>
<span id="cb46-10"><a href="genome-wide-associated-studies-gwas.html#cb46-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-11"><a href="genome-wide-associated-studies-gwas.html#cb46-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot p-values for all models</span></span>
<span id="cb46-12"><a href="genome-wide-associated-studies-gwas.html#cb46-12" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(res)</span>
<span id="cb46-13"><a href="genome-wide-associated-studies-gwas.html#cb46-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-14"><a href="genome-wide-associated-studies-gwas.html#cb46-14" aria-hidden="true" tabindex="-1"></a><span class="co"># whole genome association - one log model</span></span>
<span id="cb46-15"><a href="genome-wide-associated-studies-gwas.html#cb46-15" aria-hidden="true" tabindex="-1"></a>resHapMap <span class="ot">&lt;-</span> <span class="fu">WGassociation</span>(protein, <span class="at">data=</span> mySNP, <span class="at">model=</span><span class="st">&#39;log&#39;</span>)</span>
<span id="cb46-16"><a href="genome-wide-associated-studies-gwas.html#cb46-16" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(resHapMap)</span>
<span id="cb46-17"><a href="genome-wide-associated-studies-gwas.html#cb46-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-18"><a href="genome-wide-associated-studies-gwas.html#cb46-18" aria-hidden="true" tabindex="-1"></a>Another examplw <span class="cf">for</span> all genome association</span>
<span id="cb46-19"><a href="genome-wide-associated-studies-gwas.html#cb46-19" aria-hidden="true" tabindex="-1"></a><span class="co"># two population groups (CEU and YRI), 60 samples for each group</span></span>
<span id="cb46-20"><a href="genome-wide-associated-studies-gwas.html#cb46-20" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(HapMap)</span>
<span id="cb46-21"><a href="genome-wide-associated-studies-gwas.html#cb46-21" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(HapMap)</span>
<span id="cb46-22"><a href="genome-wide-associated-studies-gwas.html#cb46-22" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(HapMap.SNPs.pos)</span>
<span id="cb46-23"><a href="genome-wide-associated-studies-gwas.html#cb46-23" aria-hidden="true" tabindex="-1"></a><span class="co"># SNP class object</span></span>
<span id="cb46-24"><a href="genome-wide-associated-studies-gwas.html#cb46-24" aria-hidden="true" tabindex="-1"></a>myHapMap <span class="ot">&lt;-</span> <span class="fu">setupSNP</span>(HapMap, <span class="at">colSNPs=</span><span class="dv">3</span><span class="sc">:</span><span class="dv">9307</span>, <span class="at">sort=</span><span class="cn">TRUE</span>, <span class="at">info=</span>HapMap.SNPs.pos, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb46-25"><a href="genome-wide-associated-studies-gwas.html#cb46-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-26"><a href="genome-wide-associated-studies-gwas.html#cb46-26" aria-hidden="true" tabindex="-1"></a><span class="co"># association for dominant model</span></span>
<span id="cb46-27"><a href="genome-wide-associated-studies-gwas.html#cb46-27" aria-hidden="true" tabindex="-1"></a>myHapMapres <span class="ot">&lt;-</span> <span class="fu">WGassociation</span>(group, <span class="at">data=</span> myHapMap, <span class="at">model=</span><span class="st">&quot;dominant&quot;</span>)</span>
<span id="cb46-28"><a href="genome-wide-associated-studies-gwas.html#cb46-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-29"><a href="genome-wide-associated-studies-gwas.html#cb46-29" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(myHapMapres)</span>
<span id="cb46-30"><a href="genome-wide-associated-studies-gwas.html#cb46-30" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(myHapMapres)</span>
<span id="cb46-31"><a href="genome-wide-associated-studies-gwas.html#cb46-31" aria-hidden="true" tabindex="-1"></a><span class="co"># plot association for all chromosomes</span></span>
<span id="cb46-32"><a href="genome-wide-associated-studies-gwas.html#cb46-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-33"><a href="genome-wide-associated-studies-gwas.html#cb46-33" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(myHapMapres, <span class="at">whole=</span><span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="gwas-using-plink" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> GWAS using PLINK</h2>
<p>The PLINK format of the GWAS data consists of two separate files, one containing the <em>SNP information</em> (.ped)and the other containing the <em>mapping information</em> (.map). For dependence analysis, it can be combined with the phenotype data separately.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="differential-expression-analysis-using-basic-r.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="hi-c-workflow.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/03_gwas.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bioinfbook.pdf", "bioinfbook.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
