<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Hi-C workflow | A Minimal Book Example</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Hi-C workflow | A Minimal Book Example" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Hi-C workflow | A Minimal Book Example" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="Yihui Xie" />


<meta name="date" content="2021-04-16" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="genome-wide-associated-studies-gwas.html"/>

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a></li>
<li class="chapter" data-level="2" data-path="analyis-of-new-generation-sequencing-data-in-r.html"><a href="analyis-of-new-generation-sequencing-data-in-r.html"><i class="fa fa-check"></i><b>2</b> Analyis of new generation sequencing data in R</a><ul>
<li class="chapter" data-level="2.1" data-path="analyis-of-new-generation-sequencing-data-in-r.html"><a href="analyis-of-new-generation-sequencing-data-in-r.html#get-fastq-from-sra"><i class="fa fa-check"></i><b>2.1</b> Get fastq from SRA</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="get-fastq-from-sra-database.html"><a href="get-fastq-from-sra-database.html"><i class="fa fa-check"></i><b>3</b> Get fastq from SRA database</a><ul>
<li class="chapter" data-level="3.1" data-path="get-fastq-from-sra-database.html"><a href="get-fastq-from-sra-database.html#get-fastq-from-sra-database-1"><i class="fa fa-check"></i><b>3.1</b> Get fastq from SRA database</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="differential-expression-analysis-using-basic-r.html"><a href="differential-expression-analysis-using-basic-r.html"><i class="fa fa-check"></i><b>4</b> Differential expression analysis using basic R</a><ul>
<li class="chapter" data-level="4.1" data-path="differential-expression-analysis-using-basic-r.html"><a href="differential-expression-analysis-using-basic-r.html#gene-expression-analysis-of-histone-deacetylase-1-hdac1-knockout-mouse."><i class="fa fa-check"></i><b>4.1</b> Gene expression analysis of histone deacetylase 1 (HDAC1) knockout mouse.</a></li>
<li class="chapter" data-level="4.2" data-path="differential-expression-analysis-using-basic-r.html"><a href="differential-expression-analysis-using-basic-r.html#sources"><i class="fa fa-check"></i><b>4.2</b> Sources</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="genome-wide-associated-studies-gwas.html"><a href="genome-wide-associated-studies-gwas.html"><i class="fa fa-check"></i><b>5</b> Genome Wide Associated Studies (GWAS)"</a><ul>
<li class="chapter" data-level="5.1" data-path="genome-wide-associated-studies-gwas.html"><a href="genome-wide-associated-studies-gwas.html#snp-analysis-using-snpasoc-r-package"><i class="fa fa-check"></i><b>5.1</b> SNP analysis using SNPasoc R package</a></li>
<li class="chapter" data-level="5.2" data-path="genome-wide-associated-studies-gwas.html"><a href="genome-wide-associated-studies-gwas.html#gwas-using-plink"><i class="fa fa-check"></i><b>5.2</b> GWAS using PLINK</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="hi-c-workflow.html"><a href="hi-c-workflow.html"><i class="fa fa-check"></i><b>6</b> Hi-C workflow</a><ul>
<li class="chapter" data-level="6.1" data-path="hi-c-workflow.html"><a href="hi-c-workflow.html#using-straw-tool-to-extract-contact-matrix-from-.hic-files"><i class="fa fa-check"></i><b>6.1</b> Using straw tool to extract contact matrix from .hic files</a></li>
<li class="chapter" data-level="6.2" data-path="hi-c-workflow.html"><a href="hi-c-workflow.html#hic-pro-pipeline"><i class="fa fa-check"></i><b>6.2</b> HiC-Pro pipeline</a></li>
<li class="chapter" data-level="6.3" data-path="hi-c-workflow.html"><a href="hi-c-workflow.html#multihic-compare-for-compare-of-2-hi-c-datasets-rao-et-al.-2017"><i class="fa fa-check"></i><b>6.3</b> multiHiC compare for compare of 2 Hi-C datasets (Rao et al. 2017)</a></li>
<li class="chapter" data-level="6.4" data-path="hi-c-workflow.html"><a href="hi-c-workflow.html#joint-normalization-of-hi-c"><i class="fa fa-check"></i><b>6.4</b> Joint normalization of Hi-C</a></li>
<li class="chapter" data-level="6.5" data-path="hi-c-workflow.html"><a href="hi-c-workflow.html#difference-detection"><i class="fa fa-check"></i><b>6.5</b> Difference detection</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A Minimal Book Example</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="hi-c-workflow" class="section level1">
<h1><span class="header-section-number">Chapter 6</span> Hi-C workflow</h1>
<p>Workflow from Stansfield et al., Current Protocols in Bioinformatics, 2019</p>
<p>Hi-C workflow steps:<br />
1. mapping reads<br />
2. assigning fragments<br />
3. filtering fragments<br />
4. binning<br />
5. bin-level filtering<br />
6. balancing (normalization) of individual matrices</p>
<p>Paired-end reads of Hi-C experiments are mapped using the single-end mode to map each read (of the pair) independently.<br />
The theoretical maximum resolution of Hi-C sequencing is set by the restriction enzyme used to cut the DNA. However, most Hi-C datasets are not sequenced deeply enough to reach this theoretical maximum, and typically one of a few fixed-size resolutions are chosen for analyzing the data, including 1 Mb, 100 kb, 50 kb, 40 kb, 20 kb, 10 kb, and 5 kb.<br />
Two of the more popular pipelines for aligning Hi-C data are <strong>juicer</strong> (Durand, Shamim, &amp; Aiden, 2016) and <strong>HiC-Pro</strong> (Servant et al., 2015).</p>
<p><a href="https://github.com/aidenlab/juicer/wiki"><strong>juicer</strong></a> takes fastq files and aligns the data into .hic sparse contact maps. Alignment is based on BWA. Contact maps can be extracted from .hic files using juicer or the command line tool <strong>straw</strong>.</p>
<div id="using-straw-tool-to-extract-contact-matrix-from-.hic-files" class="section level2">
<h2><span class="header-section-number">6.1</span> Using straw tool to extract contact matrix from .hic files</h2>
<div class="sourceCode" id="cb47"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb47-1" title="1"><span class="op">&lt;</span><span class="ex">NONE/VC/VC_SQRT</span>/<span class="ex">KR</span><span class="op">&gt;</span> <span class="op">&lt;</span>hicFile(s)<span class="op">&gt;</span> <span class="op">&lt;</span><span class="ex">chr1</span><span class="op">&gt;</span>[:x1:x2] <span class="op">&lt;</span>chr2<span class="op">&gt;</span>[:y1:y2] <span class="op">&lt;</span>BP/FRAG<span class="op">&gt;</span> <span class="op">&lt;</span>binsize<span class="op">&gt;</span></a></code></pre></div>
<p>VC - vanilla coverage normalization<br />
VC_SQRT - square root of vanilla coverage normalization<br />
KR - Knight-Ruiz (KR) normalization<br />
BP/FRAG - base pare or fragment size. Typically we use BP.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb48-1" title="1"><span class="fu">mkdir</span> ~/GEO</a>
<a class="sourceLine" id="cb48-2" title="2"><span class="bu">cd</span> ~/GEO</a>
<a class="sourceLine" id="cb48-3" title="3"><span class="co"># download .hic file [9 Gb]</span></a>
<a class="sourceLine" id="cb48-4" title="4"><span class="fu">wget</span> ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE63nnn/GSE63525/suppl/GSE63525_K562_combined_30.hic</a>
<a class="sourceLine" id="cb48-5" title="5"><span class="co"># extract raw matrix for 22 chromosome at 500-kb resolution</span></a>
<a class="sourceLine" id="cb48-6" title="6"><span class="ex">straw</span> NONE GSE63525_K562_combined_30.hic 22 22 BP 500000 <span class="op">&gt;</span> K562.chHCT116_r22.500 kb.txt</a></code></pre></div>
<p>txt file is in the sparse upper-triangular matrix format, containing 3 columns:<br />
1. start of interaction<br />
2. end of interaction<br />
3. frequency (IF)</p>
</div>
<div id="hic-pro-pipeline" class="section level2">
<h2><span class="header-section-number">6.2</span> HiC-Pro pipeline</h2>
<p>Output 2 files (.matrix and .bed):<br />
* The .matrix is plain-text 3-column sparse upper-triangular matrix with the columns <span class="math inline">\(bin_i\)</span> <span class="math inline">\(bin_j\)</span> and <span class="math inline">\(counts_ij\)</span>.<br />
* The .bed file contains the genomic coordinates for each <span class="math inline">\(bin_i\)</span> and <span class="math inline">\(bin_j\)</span>.</p>
<p><code>sparse2full</code> - convert sparse upper-triangular matrix to full contact matrix.<br />
<code>hicpro2bedpe</code> - convert alignments by HiC-Pro into BEDPE format.<br />
<code>hicpro2bedpe</code> - input .matrix and .bed and convert into sparse upper-triangular matrix.</p>
</div>
<div id="multihic-compare-for-compare-of-2-hi-c-datasets-rao-et-al.-2017" class="section level2">
<h2><span class="header-section-number">6.3</span> multiHiC compare for compare of 2 Hi-C datasets (Rao et al. 2017)</h2>
<div class="sourceCode" id="cb49"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb49-1" title="1"><span class="fu">mkdir</span> ~/GEO</a>
<a class="sourceLine" id="cb49-2" title="2"><span class="bu">cd</span> ~/GEO</a>
<a class="sourceLine" id="cb49-3" title="3"><span class="co"># download .hic files from</span></a>
<a class="sourceLine" id="cb49-4" title="4"><span class="fu">wget</span> ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2795nnn/GSM2795535/suppl/GSM2795535_Rao-2017-HIC001_30.hic</a>
<a class="sourceLine" id="cb49-5" title="5"><span class="fu">wget</span> ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2795nnn/GSM2795536/suppl/GSM2795536_Rao-2017-HIC002_30.hic</a>
<a class="sourceLine" id="cb49-6" title="6"><span class="fu">wget</span> ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2809nnn/GSM2809539/suppl/GSM2809539_Rao-2017-HIC008_30.hic</a>
<a class="sourceLine" id="cb49-7" title="7"><span class="fu">wget</span> ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2809nnn/GSM2809540/suppl/GSM2809540_Rao-2017-HIC009_30.hic</a>
<a class="sourceLine" id="cb49-8" title="8"></a>
<a class="sourceLine" id="cb49-9" title="9"><span class="co"># convert .hic to sparse contact matrix for each of 22 chromosomes</span></a>
<a class="sourceLine" id="cb49-10" title="10"><span class="kw">for</span> <span class="ex">i</span> in <span class="dt">{1,2,8,9}</span></a>
<a class="sourceLine" id="cb49-11" title="11">   <span class="kw">do</span></a>
<a class="sourceLine" id="cb49-12" title="12">     <span class="fu">mkdir</span> HIC00<span class="va">${i}</span></a>
<a class="sourceLine" id="cb49-13" title="13">     <span class="kw">for</span> <span class="ex">j</span> in <span class="dt">{1..22}</span></a>
<a class="sourceLine" id="cb49-14" title="14">       <span class="kw">do</span></a>
<a class="sourceLine" id="cb49-15" title="15">         <span class="ex">straw</span> NONE *_Rao-2017-HIC00<span class="va">${i}</span>_30.hic <span class="va">$j</span> <span class="va">$j</span> BP 100000 <span class="op">&gt;</span> HIC00<span class="va">${i}</span>/HIC00<span class="va">${i}</span>.NONE.chr<span class="va">${j}</span>.100000.txt</a>
<a class="sourceLine" id="cb49-16" title="16">       <span class="kw">done</span></a>
<a class="sourceLine" id="cb49-17" title="17">   <span class="kw">done</span></a></code></pre></div>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb50-1" title="1"><span class="kw">library</span>(readr)</a>
<a class="sourceLine" id="cb50-2" title="2"><span class="kw">library</span>(data.table)</a>
<a class="sourceLine" id="cb50-3" title="3"><span class="kw">library</span>(dplyr)</a>
<a class="sourceLine" id="cb50-4" title="4"><span class="kw">library</span>(edgeR)</a>
<a class="sourceLine" id="cb50-5" title="5"><span class="kw">library</span>(BiocParallel)</a>
<a class="sourceLine" id="cb50-6" title="6"><span class="kw">library</span>(HiCcompare)</a>
<a class="sourceLine" id="cb50-7" title="7"><span class="kw">library</span>(multiHiCcompare)</a>
<a class="sourceLine" id="cb50-8" title="8"></a>
<a class="sourceLine" id="cb50-9" title="9"></a>
<a class="sourceLine" id="cb50-10" title="10"><span class="kw">options</span>(<span class="dt">scipen =</span> <span class="dv">10</span>) <span class="co"># output fixed numbers</span></a>
<a class="sourceLine" id="cb50-11" title="11"></a>
<a class="sourceLine" id="cb50-12" title="12"><span class="co"># Set up parameters for reading in data</span></a>
<a class="sourceLine" id="cb50-13" title="13">chr &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&#39;chr&#39;</span>, <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">22</span>))     <span class="co"># Chromosome names</span></a>
<a class="sourceLine" id="cb50-14" title="14">samples &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&#39;HIC00&#39;</span>, <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">8</span>,<span class="dv">9</span>)) <span class="co"># Sample names</span></a>
<a class="sourceLine" id="cb50-15" title="15">res &lt;-<span class="st"> </span><span class="dv">100000</span>                          <span class="co"># Data resolution</span></a>
<a class="sourceLine" id="cb50-16" title="16"></a>
<a class="sourceLine" id="cb50-17" title="17"><span class="co"># Read data</span></a>
<a class="sourceLine" id="cb50-18" title="18">sample_list &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb50-19" title="19">chr_list &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb50-20" title="20">wd &lt;-<span class="st"> &#39;/home/suvar/GEO/&#39;</span></a>
<a class="sourceLine" id="cb50-21" title="21"><span class="cf">for</span>(j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(samples)) {</a>
<a class="sourceLine" id="cb50-22" title="22">  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(chr)) {</a>
<a class="sourceLine" id="cb50-23" title="23">    chr_list[[i]] &lt;-<span class="st"> </span><span class="kw">read_tsv</span>(<span class="kw">paste0</span>(wd, samples[j], <span class="st">&#39;/&#39;</span>,</a>
<a class="sourceLine" id="cb50-24" title="24">    samples[j], <span class="st">&#39;.NONE.&#39;</span>, chr[i], <span class="st">&#39;.&#39;</span>, res, <span class="st">&#39;.txt&#39;</span>),</a>
<a class="sourceLine" id="cb50-25" title="25">    <span class="dt">col_names =</span> <span class="ot">FALSE</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.table</span>()</a>
<a class="sourceLine" id="cb50-26" title="26">    <span class="co"># Add column indicating the chromosome</span></a>
<a class="sourceLine" id="cb50-27" title="27">    chr_list[[i]] &lt;-<span class="st"> </span><span class="kw">cbind</span>(i, chr_list[[i]])</a>
<a class="sourceLine" id="cb50-28" title="28">    <span class="kw">colnames</span>(chr_list[[i]]) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;chr&#39;</span>, <span class="st">&#39;region1&#39;</span>, <span class="st">&#39;region2&#39;</span>, <span class="st">&#39;IF&#39;</span>)</a>
<a class="sourceLine" id="cb50-29" title="29">  }</a>
<a class="sourceLine" id="cb50-30" title="30">  sample_list[[j]] &lt;-<span class="st"> </span>chr_list</a>
<a class="sourceLine" id="cb50-31" title="31">  chr_list &lt;-<span class="st"> </span><span class="kw">list</span>()</a>
<a class="sourceLine" id="cb50-32" title="32">}</a>
<a class="sourceLine" id="cb50-33" title="33"></a>
<a class="sourceLine" id="cb50-34" title="34"><span class="co"># Collapse separate chromosome lists into one table per sample</span></a>
<a class="sourceLine" id="cb50-35" title="35">sample_list &lt;-<span class="st"> </span><span class="kw">lapply</span>(sample_list, rbindlist)</a>
<a class="sourceLine" id="cb50-36" title="36">  </a>
<a class="sourceLine" id="cb50-37" title="37">sample_list[[<span class="dv">1</span>]]</a></code></pre></div>
</div>
<div id="joint-normalization-of-hi-c" class="section level2">
<h2><span class="header-section-number">6.4</span> Joint normalization of Hi-C</h2>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb51-1" title="1"><span class="kw">library</span>(pander)</a>
<a class="sourceLine" id="cb51-2" title="2"><span class="co"># Create a Hicexp object for use by multiHiCcompare (~10 min)</span></a>
<a class="sourceLine" id="cb51-3" title="3"><span class="co"># Four objects are assigned into two groups</span></a>
<a class="sourceLine" id="cb51-4" title="4">rao2017 &lt;-<span class="st"> </span><span class="kw">make_hicexp</span>(<span class="dt">data_list =</span> sample_list, <span class="dt">groups =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">2</span>))</a>
<a class="sourceLine" id="cb51-5" title="5">rao2017 <span class="co"># class(rao2017)</span></a>
<a class="sourceLine" id="cb51-6" title="6"></a>
<a class="sourceLine" id="cb51-7" title="7"><span class="co"># view the IF information</span></a>
<a class="sourceLine" id="cb51-8" title="8"><span class="kw">hic_table</span>(rao2017)</a>
<a class="sourceLine" id="cb51-9" title="9"></a>
<a class="sourceLine" id="cb51-10" title="10"><span class="co"># MD plots before normalization</span></a>
<a class="sourceLine" id="cb51-11" title="11"><span class="kw">MD_hicexp</span>(rao2017, <span class="dt">plot.chr =</span> <span class="dv">1</span>, <span class="dt">plot.loess =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb52-1" title="1"><span class="co"># Normalize (~2 min)</span></a>
<a class="sourceLine" id="cb52-2" title="2">rao2017 &lt;-<span class="st"> </span><span class="kw">fastlo</span>(rao2017) <span class="co"># cyclic loesss normalisation is also available</span></a>
<a class="sourceLine" id="cb52-3" title="3"></a>
<a class="sourceLine" id="cb52-4" title="4"><span class="co"># Plot normalization results</span></a>
<a class="sourceLine" id="cb52-5" title="5"><span class="kw">MD_hicexp</span>(rao2017, <span class="dt">plot.chr =</span> <span class="dv">1</span>, <span class="dt">plot.loess =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb52-6" title="6"></a>
<a class="sourceLine" id="cb52-7" title="7"><span class="co"># Print normalized IFs</span></a>
<a class="sourceLine" id="cb52-8" title="8">pander<span class="op">::</span><span class="kw">pandoc.table</span>(<span class="kw">head</span>(<span class="kw">hic_table</span>(rao2017)))</a>
<a class="sourceLine" id="cb52-9" title="9"></a>
<a class="sourceLine" id="cb52-10" title="10"><span class="kw">library</span>(BiocParallel)</a>
<a class="sourceLine" id="cb52-11" title="11"><span class="co"># Check how many cores are available</span></a>
<a class="sourceLine" id="cb52-12" title="12">numCores &lt;-<span class="st"> </span>parallel<span class="op">::</span><span class="kw">detectCores</span>()</a>
<a class="sourceLine" id="cb52-13" title="13"><span class="co"># Set the number of cores at least one less than the total number</span></a>
<a class="sourceLine" id="cb52-14" title="14"></a>
<a class="sourceLine" id="cb52-15" title="15"><span class="cf">if</span>(<span class="kw">Sys.info</span>()[<span class="st">&#39;sysname&#39;</span>] <span class="op">==</span><span class="st"> &#39;Windows&#39;</span>) {</a>
<a class="sourceLine" id="cb52-16" title="16">  <span class="co"># Windows settings</span></a>
<a class="sourceLine" id="cb52-17" title="17">  <span class="kw">register</span>(<span class="kw">SnowParam</span>(<span class="dt">workers =</span> numCores<span class="dv">-1</span>),</a>
<a class="sourceLine" id="cb52-18" title="18">  <span class="dt">default =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb52-19" title="19">  }<span class="cf">else</span> {</a>
<a class="sourceLine" id="cb52-20" title="20">  <span class="co"># Unix settings</span></a>
<a class="sourceLine" id="cb52-21" title="21">  <span class="kw">register</span>(<span class="kw">MulticoreParam</span>(<span class="dt">workers =</span> numCores<span class="dv">-1</span>),</a>
<a class="sourceLine" id="cb52-22" title="22">  <span class="dt">default =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb52-23" title="23">  }</a></code></pre></div>
</div>
<div id="difference-detection" class="section level2">
<h2><span class="header-section-number">6.5</span> Difference detection</h2>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb53-1" title="1"><span class="co"># Perform exact test (~10 min)</span></a>
<a class="sourceLine" id="cb53-2" title="2"><span class="co"># May use &quot;parallel = TRUE&quot; option to speed up computations</span></a>
<a class="sourceLine" id="cb53-3" title="3">rao2017 &lt;-<span class="st"> </span><span class="kw">hic_exactTest</span>(rao2017, <span class="dt">parallel =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb53-4" title="4"><span class="co"># Plot a composite MD plot with the results of a comparison</span></a>
<a class="sourceLine" id="cb53-5" title="5"><span class="kw">MD_composite</span>(rao2017,<span class="dt">plot.chr =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb53-6" title="6"><span class="co"># Print results as a data frame</span></a>
<a class="sourceLine" id="cb53-7" title="7">pander<span class="op">::</span><span class="kw">pandoc.table</span>(<span class="kw">head</span>(<span class="kw">results</span>(rao2017)))</a>
<a class="sourceLine" id="cb53-8" title="8"></a>
<a class="sourceLine" id="cb53-9" title="9"><span class="co"># Save the Hicexp object</span></a>
<a class="sourceLine" id="cb53-10" title="10"><span class="kw">save</span>(rao2017, <span class="dt">file =</span> <span class="kw">paste0</span>(wd,<span class="st">&#39;rao2017.RDA&#39;</span>))</a>
<a class="sourceLine" id="cb53-11" title="11"><span class="co"># To start the downstream analysis</span></a>
<a class="sourceLine" id="cb53-12" title="12"><span class="co"># without re-running multiHiCcompare load the saved file</span></a>
<a class="sourceLine" id="cb53-13" title="13"><span class="co"># load(paste0(wd,&#39;rao2017.RDA&#39;))</span></a></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="genome-wide-associated-studies-gwas.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/04_hi-c_workflow.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bioinfbook.pdf", "bioinfbook.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
